code/game/machinery/dance_machine.dm

```diff
@@ -476,5 +466,17 @@
 		STOP_PROCESSING(SSobj, src)
 		dance_over()
 		playsound(src,'sound/machines/terminal_off.ogg',50,1)
-		icon_state = "disco0"
+		update_icon()
 		stop = world.time + 100
+
+
+/obj/machinery/jukebox/disco/process()
+	..()
+	if(active)
+		for(var/mob/M in rangers)
```

Comment:
Does this make ghosts and other weird mobs dance?

Reply:
yup!

Reply:
I might be being dumb but don't you need to tab `dance(M)` an additional time to have it be included in `isliving(M)` and therefore be checking for nonliving mobs?

Reply:
What Cit said. Currently you pass non livings as well while `dance()` expects `mob/living`

Reply:
I concur with the above. Please indent the dance call to avoid calling it on living mobs